### PR TITLE
Fix CustomTriggerArgumentBinding binding data

### PR DESCRIPTION
**Pull request by mathewc**
Fix for https://github.com/Azure/azure-functions-servicebus-extension/issues/44 in collaboration with @alrod.

### CODE REVIEW

File: test/Microsoft.Azure.WebJobs.Host.UnitTests/Common/HostCallTestsWithBindingData.cs [hunk 1]
```diff
@@ -14,9 +15,85 @@
...
+                trigger = NewTriggerObject(obj), // supplies k1,k2
+                k1 = 111 // overwrites trigger.k1
+            });
+
+            // Explicit bindingData takes precedence over binding data inferred from the trigger object.
+            Assert.Equal("111-x;200-y;111", result);
+        }
+
+        [Fact]
+        public async Task InvokeTrigger_CustomPocoConversion()
```
Comment (mathewc) [inline test/Microsoft.Azure.WebJobs.Host.UnitTests/Common/HostCallTestsWithBindingData.cs:43]: This is a repro test, exercising CustomTriggerArgumentBinding

File: test/Microsoft.Azure.WebJobs.Host.UnitTests/WebJobs.Host.UnitTests.csproj [hunk 2]
```diff
@@ -18,6 +18,11 @@
     <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
     <WarningsAsErrors />
   </PropertyGroup>
+  <ItemGroup>
+    <Compile Remove="Singleton\**" />
```
Comment (alrod) [inline test/Microsoft.Azure.WebJobs.Host.UnitTests/WebJobs.Host.UnitTests.csproj:22]: Why did we add this?

File: src/Microsoft.Azure.WebJobs.Host/Triggers/TriggerArgumentBinding/SimpleTriggerArgumentBinding.cs [hunk 3]
```diff
@@ -69,18 +75,55 @@ protected async Task<string> ConvertToStringAsync(TMessage eventData)
...
 
             string invokeString = await ConvertToStringAsync(eventData);
+            var valueProvider = new ConstantValueProvider(userValue, this.ElementType, invokeString);
+
+            return new TriggerData(valueProvider, bindingData);
+        }
 
-            IValueProvider valueProvider = new ConstantValueProvider(userValue, this.ElementType, invokeString);
-            var triggerData = new TriggerData(valueProvider, bindingData);
+        /// <summary>
```
Comment (alrod) [inline src/Microsoft.Azure.WebJobs.Host/Triggers/TriggerArgumentBinding/SimpleTriggerArgumentBinding.cs:94]: I like moving the methods to base class.

**Review by alrod — APPROVED** [review]
Approved with comments

File: test/Microsoft.Azure.WebJobs.Host.UnitTests/WebJobs.Host.UnitTests.csproj (hunk 2 above)
Comment (mathewc) [reply test/Microsoft.Azure.WebJobs.Host.UnitTests/WebJobs.Host.UnitTests.csproj:22]: It was an empty folder. Just cleaning it up :)

**Pull request merged**